packages/cli/test/unit/util/get-latest-version.test.ts

```diff
@@ -130,6 +130,42 @@ describe('get latest version', () => {
     );
     expect(() => getLatestVersion({ pkg: { name: '' } })).toThrow(TypeError);
   });
+
+  it('should reset notify if newer version is available', async () => {
+    // 1. seed the cache file with both a expireAt and notifyAt in the future
+    //    with an out-of-date latest version
+    await fs.mkdirs(join(cacheDir, 'package-updates'));
+    await fs.writeJSON(cacheFile, {
+      expireAt: Date.now(),
+      notifyAt: Date.now() - 60000,
+      version: '28.0.0',
+    });
+
+    // 2. get the latest version
+    let latest = getLatestVersion({
+      cacheDir,
+      pkg,
+    });
+    expect(latest).toEqual('28.0.0');
+
+    // we need to wait up to 20 seconds for the cacheFile to be updated
+    for (let i = 0; i < 80; i++) {
+      await sleep(250);
+      try {
+        const cache = await fs.readJSON(cacheFile);
+        if (cache.version !== '28.0.0') {
+          break;
+        }
+      } catch {
+        // cacheFile has not been updated yet
+      }
+    }
```

Comment:
**suggestion (soft-blocking):** throw explicit error on timeout

If the timeout expires, we already know that we've failed. Can we throw a clearer error message?

Reply:
Done!